pkg/validate/rsync/validate/source_spec_validator.go

```diff
@@ -306,6 +306,41 @@ var InvalidSyncCode = "1061"
 
 var invalidSyncBuilder = status.NewErrorBuilder(InvalidSyncCode)
 
+// InvalidSecretName reports that a secret name is invalid.
+func InvalidSecretName(errs []string, secretName string) status.Error {
+	return invalidSyncBuilder.
+		Sprintf("The managed secret name %q is invalid: %s. To fix it, update '.spec.git.secretRef.name'", secretName, strings.Join(errs, ", ")).
+		Build()
+}
+
+// MissingSecret reports that a secret was not found.
+func MissingSecret(namespaceSecretName string) status.Error {
+	return invalidSyncBuilder.
+		Sprintf("Secret %s not found: create one to allow client authentication", namespaceSecretName).
+		Build()
+}
+
+// ErrorFetchingSecret reports that there was an error getting a secret.
+func ErrorFetchingSecret(namespaceSecretName string, err error) status.Error {
+	return invalidSyncBuilder.Wrap(err).
+		Sprintf("Secret %s get failed", namespaceSecretName).
+		Build()
+}
+
+// InvalidSecret reports that the secret is invalid.
+func InvalidSecret(err error) status.Error {
```

Comment:
This should be decomposed into more specific `status.Error` (e.g. `MissingSecretKey` below) builders with the refactor of `validateSecretData`

Reply:
Done